core/commands/add.go

```diff
@@ -249,12 +248,16 @@ You can now check what blocks have been created by:
 				h = enc.Encode(output.Path.Cid())
 			}
 
-			if !dir && name != "" {
-				output.Name = name
-			} else {
+			if name != "" {
 				output.Name = path.Join(name, output.Name)
 			}
 
+			if !wrap && output.Name == "" {
```

Comment:
I switched this to check the _wrap_ flag instead of "is this a directory". Is that correct?

Reply:
I think we allowed "" only for wrapping directories, so this seems correct